### PR TITLE
refactor: replace char_u variables and functions with char

### PR DESCRIPTION
**Pull request by dundargoc**
Work on https://github.com/neovim/neovim/issues/459

### CODE REVIEW

File: src/nvim/ex_eval.c [hunk 1]
```diff
@@ -479,8 +479,7 @@ static int throw_exception(void *value, except_type_T type, char *cmdname)
   }
 
   excp->type = type;
-  excp->throw_name = (char *)vim_strsave(sourcing_name == NULL
-      ? (char_u *)"" : sourcing_name);
+  excp->throw_name = (char *)vim_strsave(sourcing_name == NULL ? (char_u *)"" : sourcing_name);
```
Comment (bfredl) [inline src/nvim/ex_eval.c:482]: `xstrdup(sourcing_name == NULL ? "" : (char *)sourcing_name)`

**Comment by bfredl** [comment]
otherwise LGTM

**Comment by dundargoc** [comment]
Done. Squash?

**Comment by justinmk** [comment]
rebase?

**Review by justinmk — APPROVED** [review]

**Comment by justinmk** [comment]
is this a known failure (macos CI)?

```
[  ERROR   ] test/functional/core/remote_spec.lua @ 131: Remote exits with error on send without server
test/helpers.lua:73: Expected objects to be the same.
Passed in:
(table: 0x011c2f2418) {
 *[1] = -3 }
Expected:
(table: 0x011c2f9e40) {
 *[1] = 2 }
```

**Comment by dundargoc** [comment]
That looks familiar, yes. Github has a "rerun failed-test" functionality now if you want to make sure.

**Pull request merged**